backend/src/main/java/com/google/sps/data/MatchPreference.java

```diff
@@ -41,4 +41,17 @@ public static MatchPreference forIntValue(int value) {
     }
     throw new IllegalStateException("Unknown enum value.");
   }
+
+  public static MatchPreference forStringValue(String value) {
+    switch (value) {
+      case "different":
+        return MatchPreference.DIFFERENT;
+      case "any":
+        return MatchPreference.ANY;
+      case "similar":
+        return MatchPreference.SIMILAR;
+      default:
+        return null;
```

Comment:
Should we throw exception here like the forIntValue()? Otherwise, anywhere you want to use MatchPreference, you have to check for null.